### DISCUSSION TITLE
How to access Body by object(pointer) , not ID ?

### DISCUSSION THREAD

**Discussion by finscn** [jrouwe/JoltPhysics.js#18] [General]
When I want to access/operate a body by bodyInterface ,  I need use body.GetID()
Is there an easy to use body  object directly ? 

I found c++ version could , but there is no API to do that in wasm version.

**Comment by jrouwe** [comment]
The idea is that most/all(?) of the operations are available through the body interface. This is done for locking purposes in a multi threaded environment. There is a way to obtain a body from a body ID which involves locking the body, that is indeed not exposed yet because it's not very compatible with how things work in JavaScript (can't put a variable on the stack).

(again if you want to expose this then feel free to send me a PR)

**Comment by finscn** [comment]
In pure js ,  `function-call` and  `finding-object in a map by a key` are expensive , when the number of bodies is very large .
and most games (mobile games & steam games) don't use multi-cores or multi-threads.
So I hop there could be a quick way to  access/operate a body-instance directly.

**Comment by jrouwe** [comment]
For other users, this is available now under ```physicsSystem.GetBodyLockInterface().TryGetBody(body_id)```.
